packages/node_modules/@ciscospark/redux-module-spaces/src/actions.js

```diff
@@ -0,0 +1,160 @@
+const TAG_ONE_ON_ONE = `ONE_ON_ONE`;
+
+/**
+ * Creates team object to be stored
+ *
+ * @param {Object} space
+ * @param {Object} teams
+ * @returns {Object} formatted space object
+ */
+
+function constructSpace(space, teams) {
+  const s = {
+    latestActivity: space.activities.items[0],
+    avatar: space.avatar,
+    displayName: teams[space.id] ? `General` : space.displayName,
+    id: space.id,
+    url: space.url,
+    lastReadableActivityDate: space.lastReadableActivityDate,
+    lastSeenActivityDate: space.lastSeenActivityDate,
+    conversationWebUrl: space.conversationWebUrl,
+    participants: space.participants.items,
+    type: space.tags.includes(TAG_ONE_ON_ONE) ? `direct` : `group`
+  };
+
+  if (space.team) {
+    const {teamColor, generalConversationUuid, id, archived} = space.team;
+    s.team = {
+      displayName: teams[generalConversationUuid],
+      color: teamColor,
+      generalConversationUuid,
+      id,
+      archived
+    };
+  }
+
+  return s;
+}
+
+/**
+ * Generate object of ids and team names from collection of spaces
+ *
+ * @param {Object} spaces
+ * @returns {Object} teams object
+ */
+
+function generateTeams(spaces) {
+  const teams = {};
+  spaces.forEach((s) => {
+    if (s.team && s.team.generalConversationUuid === s.id) {
+      teams[s.id] = s.displayName;
+    }
+  });
+  return teams;
+}
+
+
+export const STORE_SPACES = `spaces/STORE_SPACES`;
```

Comment:
Is this a new design pattern to have the const's right next to the functions? We should standardize on "all at the top" or "inline".